### PR TITLE
[expotools] Add command which will update changelog

### PR DESCRIPTION
**Pull request by lukmccall**
# Why

Part of https://github.com/expo/expo/issues/6178.

# How

- Added command which will add an changelog entry to the selected package. 

# Test Plan

- run `et add-changelog` ✅

### CODE REVIEW

**Review by tsapeta — CHANGES_REQUESTED** [review]

File: tools/expotools/src/commands/AddChangelog.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+async function action(options: ActionOptions) {
+  if (!process.env.CI) {
+    options = await checkOrAskForOptins(options);
+  }
+
+  if (
+    !options.author ||
+    !options.entry ||
+    !options.package ||
+    !options.pullRequest ||
```
[the remark this answers is not in the document]
Comment (lukmccall) [reply tools/expotools/src/commands/AddChangelog.ts:88]: Why should I remove this?

File: tools/expotools/src/Changelogs.ts [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+export enum ChangeType {
+  BREAKING_CHANGES = '🛠 Breaking changes',
+  NEW_FEATURES = '🎉 New features',
+  BUG_FIXES = '🐛 Bug fixes',
+}
+
+/**
+ * Heading name for unpublished changes.
+ */
+const UNPUBLISHED_VERSION_NAME = 'master';
```
[the remark this answers is not in the document]
Comment (lukmccall) [reply tools/expotools/src/Changelogs.ts]: It's a good idea. However, then we need to change headers in all changelog files and it is better to do it in a separate PR.

File: tools/expotools/src/Changelogs.ts [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+          entry.message[entry.message.length - 1] === '.' ? entry.message : `${entry.message}.`;
+        const newTokens: Markdown.Tokens = [
+          {
+            type: Markdown.TokenType.LIST_ITEM_START,
+            loose: false,
+            task: false,
+          },
+          {
+            type: Markdown.TokenType.PARAGRAPH,
+            text: `${message} ([#${entry.pullRequest}](http://github.com/expo/expo/pull/${entry.pullRequest}) by [@${entry.author}](http://github.com/${entry.author}))`,
```
Comment (tsapeta) [inline tools/expotools/src/Changelogs.ts:103]:
```suggestion
            text: `${message} ([#${entry.pullRequest}](https://github.com/expo/expo/pull/${entry.pullRequest}) by [@${entry.author}](https://github.com/${entry.author}))`,
```

File: tools/expotools/src/commands/AddChangelog.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (tsapeta) [reply tools/expotools/src/commands/AddChangelog.ts:88]: Sometimes changes are pushed directly to master - it's not good if that happens, but I can imagine a case where we want do add changelog entry without PR number through this script 😉 So imho it should be optional.

**Review by tsapeta — APPROVED** [review]

**Review by tsapeta — APPROVED** [review]

**Pull request merged**